google/ggrc-core
# GGRC-3632,4195,4196 Proposal issue

# Issue description

- add proposal engine to control and risk
- add revert revision engine to control and risk 
- send emails once in the 30 minutes 
- send email about proposals to required roles 

# Steps to test the changes

 - create pull of proposals to control or risk 
 - approve, decline or don't touch some of that proposals 
 - check that you receive right email
 - check proposals applied in right way 
 - check proposals declined in right way 
 - check comment generation in right way 
# Solution description

FE get and send data in revision content style. 
Create util that will build diff based on revision content like trees.
Generate diff based on revision contents and send it to the FE.
Generate proposal that will collect that diff. 
Apply that diff than it's needed (on apply proposal). 

Create separate cron endpoint for every 30 minutes tasks. 
Sign daily emails and proposal email to that endpoint. 
send email for required person about proposals. 

Generate requried comments on each proposal status change over signal hooks.

# Sanity checklist




# PR Review checklist
